### ISSUE TITLE
npm 2.14.1 compat

### ISSUE THREAD

**Issue by Nibbler999** [mscdex/node-mariasql#102]
As of npm 2.14.1 you can't ship a config.gypi any more. This causes installation to fail

```
[nibbler@nibbler ~]$ npm install mscdex/node-mariasql
|
> mariasql@0.1.23 install /home/nibbler/node_modules/mariasql
> node-gyp rebuild

gyp: deps/libmariadbclient/config/config.gypi not found (cwd: /home/nibbler/node_modules/mariasql) while reading includes of binding.gyp while trying to load binding.gyp
```

**Comment by Leonidaz** [comment]
@Nibbler999 were you able to resolve this?  npm just doesn't download this file.  is that just the name of the file?  it's not generated by npm / node-gyp

**Comment by Leonidaz** [comment]
ok, yes, it just doesn't like the name... i renamed the file and references to it and it started to work.  This is the name of the file that npm automatically generated and there was a security flaw and they just decided to probably remove all these files from the package..... :)

**Comment by mscdex** [comment]
FWIW, a PR to fix this is [here](https://github.com/npm/fstream-npm/pull/14). `fstream-npm` is a being bit overzealous IMHO.

**Comment by mscdex** [comment]
I also haven't noticed this on CI, which do use modern versions of npm. I thought this just affected publishing and not installing?

**Comment by Nibbler999** [comment]
The `npm install <github url>` method builds the package locally and then installs it so it is affected the same way as publishing. CI doesn't test this.

**Comment by Leonidaz** [comment]
the only other work around I found is to download the archive from Github or clone in git
place the folder into your project's "node_modules" folder, rename it to "mariasql" it it's not
in terminal cd to this folder and do "npm install"

**Comment by blah238** [comment]
:+1: think I am running into this as well: https://travis-ci.org/tgriesser/knex/jobs/86534282#L233

**Comment by mscdex** [comment]
Good news everyone! The changes to fix this have been merged into `fstream-npm`. So hopefully we will see the fix in an npm version soon.

**Comment by mscdex** [comment]
To follow up, the aforementioned changes are available in npm as of both 2.14.10 and 3.4.0.
